### PR TITLE
Improve the component actions page and related examples.

### LINKED ISSUE #1743: Document more about how zarf handles modifications to actions
### Describe what should be investigated or refactored
Currently we don't have any documentation on `./zarf` or how Zarf modifies variables and commands to run on Windows systems.  This should be added to our docs page as tips admonitions so that they are more discoverable.

### Links to any relevant code
https://docs.zarf.dev/docs/create-a-zarf-package/component-actions
https://github.com/defenseunicorns/zarf/blob/main/packages/gitea/zarf.yaml#L53

### Additional context
`./zarf` is used a lot internally and is pretty powerful to not be documenting

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This improves the component actions docs page and related examples.

## Related Issue

Fixes #1743 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: examples/kiwix/manifests/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    spec:
+      # Kiwix can hot-load files from the filesystem, but if your application cannot, this example shows how you can use an initContainer to bootstrap the injected files.
+      initContainers:
+        - name: data-loader
+          image: alpine:3.18
+          command:
+            [
+              "sh",
+              "-c",
+              'while [ ! -f /data/###ZARF_DATA_INJECTION_MARKER### ]; do echo "waiting for zarf data sync" && sleep 1; done; echo "we are done waiting!"',
```
Comment (Noxsios) [inline examples/kiwix/manifests/deployment.yaml:25]: What is the `DATA_INJECTION_MARKER`?

File: examples/kiwix/manifests/persistence.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: kiwix-data
+spec:
+  storageClassName: "###ZARF_STORAGE_CLASS###"
```
Comment (Noxsios) [inline examples/kiwix/manifests/persistence.yaml:6]: Where does `STORAGE_CLASS` come from?

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: examples/kiwix/manifests/deployment.yaml (hunk 1 above)
Comment (Racer159) [reply examples/kiwix/manifests/deployment.yaml:26]: Added a comment explaining it

File: examples/kiwix/manifests/persistence.yaml (hunk 2 above)
Comment (Racer159) [reply examples/kiwix/manifests/persistence.yaml]: This is a built in var - I copied it in from @jeff-mccoy 's original Kiwix example, but this should be a regular variable instead.

Comment (Racer159) [reply examples/kiwix/manifests/persistence.yaml:7]: NVM I forgot that this comes from `init` - basically init queries the k8s distro and selections a storage class or (optionally) it uses the override from the --storage-class flag.

https://github.com/defenseunicorns/zarf/blob/main/src/internal/cluster/state.go#L120

Comment (Racer159) [reply examples/kiwix/manifests/persistence.yaml:7]: Made that a bit clearer in a comment

**Review by YrrepNoj — DISMISSED** [review]
I think these put our docs / examples a lot more clear

**Review by YrrepNoj — APPROVED** [review]

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**